xdsl/dialects/scf.py

```diff
@@ -536,6 +536,45 @@ def __init__(
     def get(cond: SSAValue | Operation, *output_ops: SSAValue | Operation) -> Condition:
         return Condition(cond, *output_ops)
 
+    def print(self, printer: Printer):
+        printer.print("(", self.cond, ") ")
+        if self.attributes:
+            printer.print_op_attributes(self.attributes)
+            printer.print(" ")
+        if self.arguments:
+            printer.print_list(self.arguments, printer.print_ssa_value)
+            printer.print_string(" : ")
+            printer.print_list(
+                self.arguments, lambda val: printer.print_attribute(val.type)
+            )
+
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        parser.parse_punctuation("(")
+        unresolved_cond = parser.parse_unresolved_operand("cond expected")
+        parser.parse_punctuation(")")
+        cond = parser.resolve_operand(unresolved_cond, IntegerType(1))
+        attrs = parser.parse_optional_attr_dict()
+
+        # The optional list of ssa values can be confused with the list of results of the
+        # next operation. We save the position to backtrack in case we can't parse the
+        # types.
```

Comment:
This is so wrong at so many levels, but I'm starting to understand why this works in MLIR.
You are actually not allowed to backtrack, so that's not how this is parsed. This is just parsed greedily, and if there is an SSAValue then it will try to parse everything. However, since `scf.condition` is technically a terminator, you will never be able to print custom format with an `scf.condition` followed by an operation.
This is a bit broken, but it's definitely not my priority to fix that in MLIR.

Reply:
But that's what you implemented anyway, so I would just change the comment.